File: front/src/app/modules/meeting/meeting.ts
import {CalendarEvent} from 'angular-calendar';
import {UserComponent} from '../user/user.component';
import {RoomComponent} from '../room/room.component';

export interface Meeting extends CalendarEvent {
  id: string;
  title: string;
  start: Date;
  end: Date;
  /*owner: UserComponent;
  room: RoomComponent;
  members: UserComponent[];*/
}

File: fapi/src/main/java/fapi/models/Meeting.java
package fapi.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import javax.validation.constraints.FutureOrPresent;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.Date;

@Data
public class Meeting {

    private long idMeeting;

    @NotNull
    @Size(min = 4, max = 30, message = "Title must be more 4 and less 30 symbols")
    private String title;

    @NotNull
    @FutureOrPresent
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    private Date start;

    @NotNull
    @FutureOrPresent
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    private Date end;

    @NotNull
    private User owner;

    @NotNull
    private Room room;
}
